gloo/allreduce.cc

```diff
@@ -95,6 +95,10 @@ BroadcastRangeFunction genLocalBroadcastFunction(const BufferVector& out) {
 }
 
 void allreduce(const detail::AllreduceOptionsImpl& opts) {
+  if (opts.elements == 0) {
```

Comment:
Shall we remove some sanity checks below, Such as `GLOO_ENFORCE(opts.elements > 0);` and `GLOO_ENFORCE(opts.elementSize > 0);`?

Reply:
elementSize check should stay there i think. 

removed `GLOO_ENFORCE(opts.elements > 0)` since `opts.elements` is of size_t

Reply:
LGTM